src/controllers/restful/PaymentController.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * PaymentControllers
+ * RESTful Controllers for Payments Endpoints
+ */
+
+import * as express from "express";
+import { Either, left, right } from "fp-ts/lib/Either";
+import { PPTPortTypes } from "italia-pagopa-api/dist/wsdl-lib/PagamentiTelematiciPspNodoservice/PPTPort";
+import * as uuid from "uuid";
+import { PagoPaConfig } from "../../Configuration";
+import { ControllerError } from "../../enums/ControllerError";
+import { HttpErrorStatusCode } from "../../enums/HttpErrorStatusCode";
+import * as PaymentsService from "../../services/PaymentsService";
+import { PaymentsActivationRequest } from "../../types/controllers/PaymentsActivationRequest";
+import { PaymentsActivationResponse } from "../../types/controllers/PaymentsActivationResponse";
+import { PaymentsCheckRequest } from "../../types/controllers/PaymentsCheckRequest";
+import { PaymentsCheckResponse } from "../../types/controllers/PaymentsCheckResponse";
+import { CodiceContestoPagamento } from "../../types/PagoPaTypes";
+import * as PaymentsConverter from "../../utils/PaymentsConverter";
+import * as RestfulUtils from "../../utils/RestfulUtils";
+// Forward a payment check request from BackendApp to PagoPa
+export async function checkPaymentToPagoPa(
+  req: express.Request,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig
+): Promise<Either<ControllerError, PaymentsCheckResponse>> {
+  // Validate input
+  const errorOrPaymentsCheckRequest = PaymentsCheckRequest.decode(req.params);
+  if (errorOrPaymentsCheckRequest.isLeft()) {
+    RestfulUtils.sendErrorResponse(
+      res,
+      ControllerError.ERROR_INVALID_INPUT,
+      HttpErrorStatusCode.BAD_REQUEST
+    );
+    return left(ControllerError.ERROR_INVALID_INPUT);
```

Comment:
I see a pattern here. If you make `RestfulUtils.sendErrorResponse` return the Either (error) you can just use

`return RestfulUtils.sendErrorResponse(...)` so you don't have to repeat the error type two times (which is error prone)